sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubProducerClient.cs

```diff
@@ -427,53 +488,22 @@ public virtual async Task<string[]> GetPartitionIdsAsync(CancellationToken cance
             Argument.AssertNotNull(eventBatch, nameof(eventBatch));
             AssertSinglePartitionReference(options.PartitionId, options.PartitionKey);
 
-            int attempts = 0;
-
-            eventBatch = (eventBatch as IList<EventData>) ?? eventBatch.ToList();
-            InstrumentMessages(eventBatch);
-
-            var diagnosticIdentifiers = new List<string>();
-
-            foreach (var eventData in eventBatch)
+            var events = eventBatch switch
             {
-                if (EventDataInstrumentation.TryExtractDiagnosticId(eventData, out var identifier))
-                {
-                    diagnosticIdentifiers.Add(identifier);
-                }
-            }
-
-            using DiagnosticScope scope = CreateDiagnosticScope(diagnosticIdentifiers);
-
-            var pooledProducer = PartitionProducerPool.GetPooledProducer(options.PartitionId, PartitionProducerLifespan);
+               IReadOnlyList<EventData> eventList => eventList,
+               _ => eventBatch.ToList()
```

Comment:
I don't think we do.   This is an internal transform because we enumerate the set multiple times and want to also use the `Count`.   The read-only bits are a statement of intent rather than something we need to strictly enforce.  The docs say that it's a light wrapper, but I don't want to potentially trigger an allocation if we don't need it.

Any concern that I may be overlooking?